src/server/handlers/launchpad.js

```diff
@@ -384,6 +394,60 @@ const internalFindSnapsByPrefix = async (urlPrefix) => {
   }
 };
 
+// If we haven't yet initialized metrics related to the number of snaps for
+// this user, do that now making some reasonable assumptions about
+// historical data.  This may under-report if somebody manages to accumulate
+// more snaps than the Launchpad batch size (75) before triggering this, but
+// that shouldn't happen in practice.
+const initializeMetrics = async (gitHubId, snaps) => {
+  await db.transaction(async (trx) => {
+    const row = await db.model('GitHubUser')
+      .where({ github_id: gitHubId })
+      .fetch({ transacting: trx });
+    if (row) {
+      const rowData = row.serialize();
+      if (rowData.snaps_added === undefined ||
+          rowData.snaps_added === null) {
+        row.set({ snaps_added: snaps.length });
+      }
+      if (rowData.snaps_removed === undefined ||
+          rowData.snaps_removed === null) {
+        row.set({ snaps_removed: 0 });
+      }
+      if (rowData.names_registered === undefined ||
+          rowData.names_registered === null) {
+        let namesRegistered = 0;
+        for (const snap of snaps) {
+          if (snap.store_name) {
+            namesRegistered += 1;
+          }
+        }
```

Comment:
Could `namesRegistered` be generated using `Array.prototype.find` and an arrow function?

```
const namesRegistered = snaps.filter((snap) => snap.store_name).length;
```